Wikipedia/Code/WMFSettingsViewController.m

```diff
@@ -387,50 +382,66 @@ - (void)tweakViewControllerPressedDone:(FBTweakViewController *)tweakViewControl
 
 - (nullable NSIndexPath *)indexPathForVisibleCellOfType:(WMFSettingsMenuItemType)type {
     return [self.tableView.indexPathsForVisibleRows wmf_match:^BOOL(NSIndexPath *indexPath) {
-        return ((WMFSettingsMenuItem *)[self.elementDataSource itemAtIndexPath:indexPath]).type == type;
+        return ([self.tableView cellForRowAtIndexPath:indexPath].tag == type);
     }];
 }
 
 - (void)reloadVisibleCellOfType:(WMFSettingsMenuItemType)type {
     NSIndexPath *indexPath = [self indexPathForVisibleCellOfType:type];
     if (indexPath) {
-        [self.elementDataSource replaceItemAtIndexPath:indexPath withItem:[WMFSettingsMenuItem itemForType:type]];
+        [self.tableView reloadRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationNone];
```

Comment:
You could probably delete this method entirely and just use reload sections instead. There probably isn't a huge performance advantage to just reloading the rows that changed.